tests/backends/aiida_sqlalchemy/test_utils.py

```diff
@@ -98,9 +98,9 @@ def create_database(url, encoding='utf8'):
             from psycopg2.extensions import ISOLATION_LEVEL_AUTOCOMMIT
             engine.raw_connection().set_isolation_level(ISOLATION_LEVEL_AUTOCOMMIT)
 
-            text = f"CREATE DATABASE {quote(engine, database)} ENCODING '{encoding}'"
-
-            engine.execute(text)
+            text = sa.text(f"CREATE DATABASE {quote(engine, database)} ENCODING '{encoding}'")
+            with engine.begin() as connection:
+                connection.execute(text)
```

Comment:
Why did this now need a context to handle the connection? Specially since we actually removed the context handling from the other place in the migrations part? (my other comment)

Reply:
these are different concepts; that other context was already in an (outer) context (i.e. transaction), so did not need an additional one, whereas this is the only context for this transaction